tests/Microsoft.Identity.Test.Integration.netfx/HeadlessTests/ManagedIdentityTests.cs

```diff
@@ -0,0 +1,189 @@
+            }
+            catch (Exception ex)
+            {
+                exception = ex.Message;
+            }
+
+            //Assert
+            Assert.IsTrue(string.IsNullOrEmpty(exception));
+            Assert.AreSame(result, expectedClientException);
+        }
+
+        [DataTestMethod]
+        [DataRow(AzureResource.webapp, "", DisplayName = "System Identity Web App")]
+        [DataRow(AzureResource.function, "", DisplayName = "System Identity Function App")]
+        [DataRow(AzureResource.webapp, UserAssignedClientID, DisplayName = "User Identity Web App")]
+        [DataRow(AzureResource.function, UserAssignedClientID, DisplayName = "User Identity Function App")]
+        [DataRow(AzureResource.webapp, Mi_res_id, DisplayName = "ResourceID Web App")]
+        [DataRow(AzureResource.function, Mi_res_id, DisplayName = "ResourceID Function App")]
+        public async Task AcquireMSITokenAsync(AzureResource azureResource, string userIdentity)
+        {
+            //Arrange
+            AuthenticationResult result = null;
+
+            //Set the Environment Variables
+            bool isEnvironmentVariableSet = await SetEnvironmentVariablesAsync(azureResource)
+                .ConfigureAwait(false);
+
+            if (!isEnvironmentVariableSet)
+            {
+                Assert.Fail("AcquireMSITokenForWebAppAsync failed to set environment variables.");
+            }
+
+            //form the URI 
+            string uri = s_baseURL + $"GetMSIToken?azureresource={azureResource}&uri=";
+
+            //Create CCA with Proxy
+            IConfidentialClientApplication cca = CreateCCAWithProxy(uri);
+
+            //Act
+            try
+            {
+                result = await cca.AcquireTokenForClient(s_msi_scopes)
+                    .WithManagedIdentity(userIdentity)
+                    .ExecuteAsync().ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                Trace.WriteLine(ex);
+                Assert.IsNull(ex.Message);
+            }
+
+            //Assert
+            Assert.IsNotNull(result.AccessToken);
+            Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+
+            //User Identity Client ID 
+            //Need to check if we are exposing Client ID in msal token response
+            //if (!string.IsNullOrEmpty(userIdentity))
+            //{
+            //    Assert.AreEqual(UserAssignedClientID, result.);
+            //}
+
+            result = await cca.AcquireTokenForClient(s_msi_scopes)
+                .WithManagedIdentity(userIdentity)
+                .ExecuteAsync().ConfigureAwait(false);
+
+            Assert.IsNotNull(result.Scopes);
+            Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
+
+            ClearEnvironmentVariables();
+
+        }
+
+        private async Task<bool> SetEnvironmentVariablesAsync(AzureResource resource)
```

Comment:
Not a fan of returning true / false. C# pattern is to use exceptions, because they convey more information.